test/services/network/tracing/traversals/test_basic_traversal.py

```diff
@@ -0,0 +1,161 @@
+#  Copyright 2022 Zeppelin Bend Pty Ltd
+#
+#  This Source Code Form is subject to the terms of the Mozilla Public
+#  License, v. 2.0. If a copy of the MPL was not distributed with this
+#  file, You can obtain one at https://mozilla.org/MPL/2.0/.
+from typing import List, Callable, Awaitable
+
+import pytest
+
+from zepben.evolve import BasicTraversal, breadth_first, Traversal
+
+
+def _queue_next(i: int, t: BasicTraversal[int]):
+    for n in [i - 2, i - 1, i + 1, i + 2]:
+        if n > 0:
+            t.process_queue.put(n)
+
+
+def _geq(n: int) -> Callable[[int], Awaitable[bool]]:
+    async def compare(i: int):
+        return i >= n
+
+    return compare
+
+
+def _append_to(a: List) -> Callable[[int, bool], Awaitable[None]]:
+    async def append(i: int, _: bool):
+        a.append(i)
+
+    return append
+
+
+@pytest.mark.asyncio
+async def test_breadth_first():
```

Comment:
todo: I do believe these tests already existed in `test_traversal.py`, incorrectly under the `TestTraversal` class. As part of the review/split of that file can you please reconcile it with what you have added here.